src/metabase/query_processor/middleware/metrics.clj

```diff
@@ -245,14 +258,18 @@
    2. Metric source cards can reference themselves.
       A query built from a `:source-card` of `:type :metric` can reference itself."
   [query]
-  (let [query (lib.walk/walk
-               query
-               (fn [_query path-type path stage-or-join]
-                 (when (= path-type :lib.walk/join)
-                   (update stage-or-join :stages #(adjust-metric-stages query path %)))))]
-    (u/prog1
-      (update query :stages #(adjust-metric-stages query nil %))
-      (when-let [metric (lib.util.match/match-one <>
-                          [:metric _ _] &match)]
-        (log/warn "Failed to replace metric"
-                  (pr-str {:metric metric}))))))
+  (if-not (match-one-metric query)
```

Comment:
@metamben N.B. this change, we no longer call `adjust-metric-stages` if we don't find a `:metric` clause (otherwise, we update the "total adjustments" counter and attempt to adjust). Looks safe to me but let me know if not.

Reply:
I expect this should be a no-op if the query isn't consuming metrics. 👍